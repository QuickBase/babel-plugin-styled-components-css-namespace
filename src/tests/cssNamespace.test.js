import pluginTester from 'babel-plugin-tester';
import plugin from '../index';
import path from 'path';

pluginTester({
  plugin,
  pluginName: 'styled-components-css-namespace',
  filename: __filename,
  // fixtures: path.join(__dirname, './fixtures'),
  snapshot: true,
  tests: [
    {
      title: 'adds namespace to simple styled-component',
      fixture: path.join(__dirname, './fixtures/simple_styled_component.js')
    },

    {
      title: 'adds namespace to complex styled-component',
      fixture: path.join(__dirname, './fixtures/complex_styled_component.js')
    },
    {
      title: 'uses a namespace specified in the options',
      pluginOptions: {
        cssNamespace: 'moreSpecific'
      },
      fixture: path.join(__dirname, './fixtures/complex_styled_component.js')
    },
    {
      title: 'uses an array of namespaces specified in the options',
      pluginOptions: {
        cssNamespace: ['specific', 'verySpecific', 'extraSpecific']
      },
      fixture: path.join(__dirname, './fixtures/complex_styled_component.js')
    },
    {
      title:
        'does not transform a template string that is not a styled component',
      snapshot: false, // snapshot should be false because code should be unmodified
      fixture: path.join(__dirname, './fixtures/not_styled_component.js')
    },
    {
      title: 'does not add extra selectors to child helper styles',
      fixture: path.join(
        __dirname,
        './fixtures/nested_helper_styled_component.js'
      )
    },
    {
      title: 'does not add namespace to keyframes',
      pluginOptions: {
        cssNamespace: 'specific'
      },
      fixture: path.join(__dirname, './fixtures/keyframes_styled_component.js')
    },
    {
      title: 'adds namespace to extended styled-components',
      fixture: path.join(__dirname, './fixtures/extended_styled_component.js')
    }
  ]
});
