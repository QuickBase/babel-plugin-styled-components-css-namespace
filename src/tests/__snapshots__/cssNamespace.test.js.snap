// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds namespace to complex styled-component 1`] = `
"
const styled = { div() {} };

const MyStyledComponent = styled.div\`
  background-color: \${props => (props.isDark ? 'red' : 'yellow')};
  color: \${props => (props.isDark ? 'white' : 'navy')};
  font-size: 30px;
  padding: 20px;

  & {
    border: 1px solid black;
  }
\`;

export default MyStyledComponent;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = { div() {} };

const MyStyledComponent = styled.div\`
&& {
  background-color: \${props => props.isDark ? 'red' : 'yellow'};
  color: \${props => props.isDark ? 'white' : 'navy'};
  font-size: 30px;
  padding: 20px;

  & {
    border: 1px solid black;
  }
}
\`;

export default MyStyledComponent;
"
`;

exports[`adds namespace to simple styled-component 1`] = `
"
const styled = { div() {} };

const MyStyledComponent = styled.div\`
  background-color: 'yellow';
\`;

export default MyStyledComponent;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = { div() {} };

const MyStyledComponent = styled.div\`
&& {
  background-color: 'yellow';
}
\`;

export default MyStyledComponent;
"
`;

exports[`does not add extra selectors to child helper styles 1`] = `
"
const styled = { div() {} };
const css = { css() {} };

const styledCSS = css\`
  width: 100px;
\`;

const MyStyledComponent = styled.div\`
  background-color: 'yellow';
  \${styledCSS};
\`;

export default MyStyledComponent;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = { div() {} };
const css = { css() {} };

const styledCSS = css\`
  width: 100px;
\`;

const MyStyledComponent = styled.div\`
&& {
  background-color: 'yellow';
  \${styledCSS};
}
\`;

export default MyStyledComponent;
"
`;

exports[`uses a namespace specified in the options 1`] = `
"
const styled = { div() {} };

const MyStyledComponent = styled.div\`
  background-color: \${props => (props.isDark ? 'red' : 'yellow')};
  color: \${props => (props.isDark ? 'white' : 'navy')};
  font-size: 30px;
  padding: 20px;

  & {
    border: 1px solid black;
  }
\`;

export default MyStyledComponent;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = { div() {} };

const MyStyledComponent = styled.div\`
.moreSpecific & {
  background-color: \${props => props.isDark ? 'red' : 'yellow'};
  color: \${props => props.isDark ? 'white' : 'navy'};
  font-size: 30px;
  padding: 20px;

  & {
    border: 1px solid black;
  }
}
\`;

export default MyStyledComponent;
"
`;

exports[`uses an array of namespaces specified in the options 1`] = `
"
const styled = { div() {} };

const MyStyledComponent = styled.div\`
  background-color: \${props => (props.isDark ? 'red' : 'yellow')};
  color: \${props => (props.isDark ? 'white' : 'navy')};
  font-size: 30px;
  padding: 20px;

  & {
    border: 1px solid black;
  }
\`;

export default MyStyledComponent;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = { div() {} };

const MyStyledComponent = styled.div\`
.specific .verySpecific .extraSpecific & {
  background-color: \${props => props.isDark ? 'red' : 'yellow'};
  color: \${props => props.isDark ? 'white' : 'navy'};
  font-size: 30px;
  padding: 20px;

  & {
    border: 1px solid black;
  }
}
\`;

export default MyStyledComponent;
"
`;
